### PR TITLE
BUG: Ignore differences in NAN for computing ULP differences

### LINKED ISSUE #15593: BUG: test_umath_accuracy tests for a specific NAN
Inside the manylinux2010 i686 docker, `log(-1)` returns `nan`, but it is a different `nan` than the one expected by the [data](https://github.com/numpy/numpy/pull/14048/files#diff-c3b4a71eff65b42180aea2630ee50a92R11) used in `test_umath_accuracy`. The tests expect `0xffc00000`, but the glibc `log()` function on this platform returns `0x7fc00000`. Both are `nan`, but [`assert_array_max_ulp`](https://github.com/numpy/numpy/pull/14048/files#diff-8df7ea06c4577eeeba7aa23db566034aR53) uses `nulp_diff` which is very literal:
```
>>> import numpy as np, numpy.testing
>>> x = np.int32(0xffc00000).view(np.float32)
>>> y = np.int32(0x7fc00000).view(np.float32)
>>> print(x, y)
nan nan
>>> numpy.testing._private.utils.nulp_diff(y, x)
8388608.0
```
`np.isnan` returns True for both.

I think the test be relaxed to accept more than one `nan`.

### PR DESCRIPTION
**Pull request by r-devulap**
Fixes https://github.com/numpy/numpy/issues/15593

### CODE REVIEW

File: numpy/core/tests/test_umath_accuracy.py [hunk 1]
```diff
@@ -50,5 +50,7 @@ def test_validate_transcendentals(self):
                         perm = np.random.permutation(len(inval))
                         inval = inval[perm]
                         outval = outval[perm]
+                        inval[np.argwhere(np.isnan(inval))] = np.nan
+                        outval[np.argwhere(np.isnan(inval))] = np.nan
```
Comment (mattip) [inline numpy/core/tests/test_umath_accuracy.py]: a comment here would be helpful

Comment (seberg) [reply numpy/core/tests/test_umath_accuracy.py]: Can we fix this within the `assert_array_max_ulp` function? That function should be consistent in either rejecting or accepting NaN I think.

Comment (mattip) [reply numpy/core/tests/test_umath_accuracy.py]: `assert_array_max_ulp`  indeed does not document its NaN behaviour, so I guess we can document and fix it there.

Comment (mattip) [reply numpy/core/tests/test_umath_accuracy.py]: also it is a bit annoying that `assert_array_max_ulp` does not show the offending output like `assert_array_compare` but only the maximum ULP discovered.

Comment (r-devulap) [reply numpy/core/tests/test_umath_accuracy.py]: makes sense, I can update the `assert_array_max_ulp ` function instead. Where would you like me to add the document?

Comment (mattip) [reply numpy/core/tests/test_umath_accuracy.py]: the docstring of the function itself does not indicate any special handling of NaNs. If we change that, we should document it

Comment (mattip) [reply numpy/core/tests/test_umath_accuracy.py]: in the docstring

**Comment by seberg** [comment]
Thanks, do you think it might make sense to push it even further down to the `integer_representation` function? Since the "signed magnitude" representation is not just a view as far as I understand it, that might make sense?

**Comment by r-devulap** [comment]
not so sure about that, it is the max ULP diff function that doesn't want to differentiate between NAN's. The `integer_representation`  function simply returns the binary representation of a floating point and I think that should stay true to its intent for NAN's as well.

**Comment by r-devulap** [comment]
`x[np.argwhere(np.isnan(x))] = np.nan` fails with `E       IndexError: too many indices for array` for a single element array like `x = np.array(1.2) ` Any suggestions on how to elegantly fix it without special casing it?

**Comment by seberg** [comment]
@r-devulap any reasons you are using `np.argwhere` at all, just remove it? Not sure I agree with the interpretation of `integer_representation`, but it also does not really matter much right now.

**Comment by r-devulap** [comment]
> @r-devulap any reasons you are using `np.argwhere` at all, just remove it? 

Thanks :)

**Comment by seberg** [comment]
@r-devulap looks good to me, would you be up for adding a test as well, since I think the ulp thing is arguably a bug fix in our testing utils, and also testing utils need testing :).

**Comment by r-devulap** [comment]
Sure, what do you have in mind? ensure ulp error between two different representations of NAN = 0?

**Comment by r-devulap** [comment]
@seberg let me know if that suffices.

**Comment by seberg** [comment]
Test would be nicer if you move it into `numpy/testing/tests/test_utils.py`. Otherwise looks good to me, I will let Matti sign off on the line removal changes and merge. Thanks!

**Comment by mattip** [comment]
I think there is more to do with the umath_accuracy test but this solves the current issues. xref gh-15605 for some possible improvements

**Comment by mattip** [comment]
Thanks @r-devulap for the quick response

**Pull request merged**